tensorflow/contrib/image/python/ops/image_ops.py

```diff
@@ -55,80 +57,155 @@ def rotate(images, angles, interpolation="NEAREST"):
   Raises:
     TypeError: If `image` is an invalid type.
   """
-  image_or_images = ops.convert_to_tensor(images, name="images")
-  if image_or_images.dtype.base_dtype not in _IMAGE_DTYPES:
-    raise TypeError("Invalid dtype %s." % image_or_images.dtype)
-  if len(image_or_images.get_shape()) == 2:
-    images = image_or_images[None, :, :, None]
-  elif len(image_or_images.get_shape()) == 3:
-    images = image_or_images[None, :, :, :]
-  elif len(image_or_images.get_shape()) == 4:
-    images = image_or_images
-  else:
-    raise TypeError("Images should have rank between 2 and 4.")
-
-  image_height = math_ops.cast(array_ops.shape(images)[1], dtypes.float32)[None]
-  image_width = math_ops.cast(array_ops.shape(images)[2], dtypes.float32)[None]
-  output = transform(
-      images,
-      angles_to_projective_transforms(angles, image_height, image_width),
-      interpolation=interpolation)
-  if len(image_or_images.get_shape()) == 2:
-    return output[0, :, :, 0]
-  elif len(image_or_images.get_shape()) == 3:
-    return output[0, :, :, :]
-  else:
-    return output
+  with ops.name_scope(name):
+    image_or_images = ops.convert_to_tensor(images)
+    if image_or_images.dtype.base_dtype not in _IMAGE_DTYPES:
+      raise TypeError("Invalid dtype %s." % image_or_images.dtype)
+    if len(image_or_images.get_shape()) == 2:
```

Comment:
Check that shape.ndims is not none first

Reply:
Done throughout.